src/platform/intel/ace/platform.c

```diff
@@ -108,6 +119,12 @@ static void notify_pm_state_entry(enum pm_state state)
 						 0,
 						 SOF_MEM_CAPS_L3,
 						 storage_buffer_size);
+
+		/* IPC driver may not allow to change power state */
+		int ret = pm_device_action_run(INTEL_ADSP_IPC_HOST_DEV, PM_DEVICE_ACTION_SUSPEND);
```

Comment:
So it looks like it can't "do it himself".
I've spend some time trying to figure this out and the registered in the IPC device [PM control function](https://docs.zephyrproject.org/2.6.0/reference/power_management/index.html#device-model-with-power-management-support) (here `ipc_pm_action` registered with the `PM_DEVICE_DT_DEFINE` macro) is not called "automatically" in any way.
This is also indicated in the Zephyr docs [Device Power Management API](https://docs.zephyrproject.org/2.6.0/reference/power_management/index.html#device-power-management-api).
Partial quote:

> Device Power Management API[¶](https://docs.zephyrproject.org/2.6.0/reference/power_management/index.html#device-power-management-api)
> The SOC interface and application use these APIs to perform power management operations on the devices.
(...)

> Device Set Power State[¶](https://docs.zephyrproject.org/2.6.0/reference/power_management/index.html#device-set-power-state)
> int pm_device_state_set(const struct device *dev, uint32_t device_power_state, pm_device_cb cb, void *arg);
> Calls the pm_control() handler function implemented by the device driver with PM_DEVICE_STATE_SET command.

So there is not magic here for the Device PM model. Device PM control functions are registered with PM_DEVICE_DT_DEFINE and added to the internal list, but this list is not invoked automatically in any way.
Documentation clearly states that we should call power state transition for the Device manually using `pm_device_state_set()` from the Zephyr application level. Besides, even if that list would be called automatically, there would be a question of the ordering (which pm control functions for which devices to call in what order).
I'm open for suggestions on how to do it automatically.

Reply:
Fixed, there was a bug in Zephyr PM, now it can do it by himself.